django/contrib/postgres/utils.py

```diff
@@ -0,0 +1,30 @@
+from __future__ import unicode_literals
+
+from django.core.exceptions import ValidationError
+from django.utils.functional import SimpleLazyObject
+from django.utils.translation import string_concat
+
+
+def prefix_validation_error(error, prefix, code, params):
+    """
+    Prefix a validation error message while maintaining the existing
+    validation data structure.
+    """
+    if error.error_list == [error]:
+        error_params = error.params or {}
+        return ValidationError(
+            # We can't simply concatenate messages since they might require
+            # their associated parameters to be expressed correctly which
+            # is not something `string_concat` does. For example, proxied
+            # ungettext calls require a count parameter and are converted
+            # to an empty string if they are missing it.
```

Comment:
This should be reworded but the main issue is `string_concat` can't be used when `error.message` is a proxied call to `u?ngettext` since the later can't be simply `force_text`ed.